### ISSUE TITLE
Running PWD in Docker Desktop for Mac

### ISSUE THREAD

**Issue by lmakarov** [play-with-docker/play-with-docker#329]
I've been playing with PWD for a while as a docker playground and I'm really fascinated by the project ❤️ 

I'm now trying to get PWD running locally in Docker Desktop for Mac to better understand it's internals.

I'm following the steps in README.

This does not seem to be applicable to Moby Linux or I'm not sure how to handle this:

> It's also necessary to manually load the IPVS kernel module because as swarms are created in dind, the daemon won't load it automatically. Run the following command for that purpose: sudo modprobe xt_ipvs

Others steps seem to be straightforward.

```
$ docker swarm init
Error response from daemon: This node is already part of a swarm. Use "docker swarm leave" to leave this swarm and join another one.
```

```
docker pull franela/dind
Using default tag: latest
latest: Pulling from franela/dind
Digest: sha256:9de94cd0fa7957ab5830112ff25586a5799b57d2da8524f0586f99c4114cfb17
Status: Image is up to date for franela/dind:latest
```

```
$ dep ensure
Warning: the following project(s) have [[constraint]] stanzas in Gopkg.toml:

  ✗  github.com/docker/distribution
  ✗  github.com/googollee/go-socket.io
  ✗  github.com/zbindenren/negroni-prometheus

However, these projects are not direct dependencies of the current project:
they are not imported in any .go files, nor are they in the 'required' list in
Gopkg.toml. Dep only applies [[constraint]] rules to direct dependencies, so
these rules will have no effect.

Either import/require packages from these projects so that they become direct
dependencies, or convert each [[constraint]] to an [[override]] to enforce rules
on these projects, if they happen to be transitive dependencies.
```

```
$ docker-compose up
WARNING: The Docker Engine you're using is running in swarm mode.

Compose does not use swarm mode to deploy services to multiple nodes in a swarm. All containers will be scheduled on the current node.

To deploy your application across the swarm, use `docker stack deploy`.

Creating network "play-with-docker_default" with the default driver
Creating haproxy ... done
Creating pwd     ... done
Creating l2      ... done
Attaching to haproxy, pwd, l2
haproxy    | [NOTICE] 094/053154 (1) : New worker #1 (6) forked
l2         | 2019/04/05 05:32:06 Connected to network [play-with-docker_default] with ip [172.22.0.4]
pwd        | 2019/04/05 05:32:50 Updating playgrounds configuration
pwd        | 2019/04/05 05:32:50 EVENT: Playground New 4a24b940-e23f-54f5-8867-137433240dd7
pwd        | 2019/04/05 05:32:50 Updating playgrounds configuration
pwd        | 2019/04/05 05:32:50 Listening on port 3000
```

Now, trying to open http://localhost and getting `ERR_EMPTY_RESPONSE ` in Chrome. Nothing shows up in the running `docker-compose up` output.
This tells me something is messed up with haproxy routing. Skipping it for now.

Updated `docker-compose.yml` to expose `pwd:3000` and bypass haproxy. Ran `docker-compose up` again. This time, getting `HTTP ERROR 404` in Chrome. docker-compose output shows this:

```
pwd        | [negroni] Started GET /
pwd        | 2019/04/05 05:39:47 Error retrieving playground 8547b5e6-08d0-5856-bae1-63c87d64d497. Got: NotFound
pwd        | 2019/04/05 05:39:47 Playground for domain localhost:3000 was not found!
pwd        | [negroni] Completed 404 Not Found in 1.335111ms
```

There seem to be multiple issues with the setup. Where do I move from here?
How do you develop PWD locally?

Note: I'm new to golang, but experienced with docker.

**Comment by marcosnils** [comment]
Hey!. Haven't tried to deploy PWD on a mac. I work on a linux machine regulary. 

I'll try to grab a mac this week and give it a try there. From the contributors list I believe @alexellis and @mikesir87 could be able to provide insights about how to set it up on a mac.

**Comment by lmakarov** [comment]
I was eventually able to get PWD up and running with Docker for Mac.

The issue (in my case) was with using `http://localhost/`.  I was constantly getting "empty response from the server" everywhere (Chrome, Firefox, curl). Then, I noticed how curl was resolving `locahost`:

```
$ curl -v http://localhost/
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 80 (#0)
> GET / HTTP/1.1
> Host: localhost
> User-Agent: curl/7.54.0
> Accept: */*
> 
* Empty reply from server
* Connection #0 to host localhost left intact
curl: (52) Empty reply from server
```

Note **localhost (::1)** (IPv6)

I commented out the default (on Mac) `::1             localhost` line in `/etc/hosts`.

Curl was now happily resolving `localhost` to `127.0.0.1` and getting an answer from haproxy and PWD. Firefox was also doing fine now. Chrome was still not happy (even after restarting it).

To get PWD to open up in Chrome, I switched from `http://locahost` to `http://pwd` (added `127.0.0.1 pwd` in `/etc/hosts`) and that worked.

I can move forward now :)
